mypaint-brush.c

```diff
@@ -714,6 +714,8 @@ smallest_angular_difference(float a, float b)
         b = self->states[MYPAINT_BRUSH_STATE_LAST_GETCOLOR_B];
         a = self->states[MYPAINT_BRUSH_STATE_LAST_GETCOLOR_A];
       }
+      //convert to RYB-- we want to mix the smudge with the new getcolor in RYB space
+      rgb_to_ryb_float (&r, &g, &b);
```

Comment:
Should it be controlled by a setting, just in case?

A single setting could do it: 0(default)=RGB, 1=RYB. Though you'd have to crossfade the RYB mix colour with the RGB one in between.